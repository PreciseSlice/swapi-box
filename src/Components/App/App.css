.app {
  min-height: 100%;
  width: 100%; }

.main {
  border: 2px solid #feda4a;
  height: 450px;
  margin: 40px 40px; }

.header {
  display: flex;
  margin: 25px;
  justify-content: center; }
  .header h1 {
    color: #feda4a;
    font-size: 5em; }
  .header button {
    margin-left: 20px; }
