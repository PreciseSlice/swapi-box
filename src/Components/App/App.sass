.app 
  min-height: 100%
  width: 100%

.main 
  height: 480px
  margin: 30px 30px

.header 
  display: flex
  margin: 25px
  justify-content: center

  h1
    color: #feda4a
    letter-spacing: 0.25em
    font-size: 5em
    font-weight: 600

  button
    margin-left: 20px 
