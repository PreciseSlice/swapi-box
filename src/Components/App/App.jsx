import React, { Component } from 'react';
import './App.css';
import Header from '../Header/Header.jsx';
import Nav from '../Nav/Nav.jsx';
import Main from '../Main/Main';
//import { Route, NavLink, Link } from 'react-router-dom';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      films: [],
      peopleData: []
    };
  }

  componentDidMount() {
    const peopleData = this.getPeople();
    const films = this.getfilms();
    this.setState({
      peopleData
    });
  }

  getFilms = async () => {
    const fetchFilmData = await fetch('https://swapi.co/api/films/');
  };

  getPeople = async () => {
    const fetchPeopleData = await fetch('https://swapi.co/api/people/');
    const cleanPeople = await fetchPeopleData.json();
    const peopleMap = cleanPeople.results.map(async person => {
      const name = person.name;
      const fetchHomeworld = await fetch(person.homeworld);
      const cleanHomeworld = await fetchHomeworld.json();

      const homeworld = cleanHomeworld.name;
      const population = cleanHomeworld.population;

      const fetchSpecies = await fetch(person.species);
      const cleanSpecies = await fetchSpecies.json();
      const speciesName = cleanSpecies.name;

      return { name, homeworld, population, speciesName };
    });
    return Promise.all(peopleMap);
  };

  render() {
    return (
      <div className="app">
        <Header />
        <Nav />
        <Main />
      </div>
    );
  }
}

export default App;

// https://swapi.co/api/films/ .results has an array with all seven films
// pick 0-7 at random, get text from ."opening_crawl"

// Footer with major styling and gif/image backgroud dark space fleet img might
// work well for the background of the whole thing.
// Second image with closer perspective on the footer

// Landing page does the scroll intro and then go to main page
// or scroll in nav and then cards render
