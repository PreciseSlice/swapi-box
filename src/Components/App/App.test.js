/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow, mount } from 'enzyme';
import {
  cleanFilms,
  cleanPeople,
  cleanPlanets,
  cleanVehicles
} from '../apiCalls/apiMockData';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('Exist and matches snapshot', () => {
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  it('Should start with a default properties', () => {
    expect(wrapper.state().filmData).toEqual({});
    expect(wrapper.state().peopleData).toEqual([]);
    expect(wrapper.state().planetData).toEqual([]);
    expect(wrapper.state().vehicleData).toEqual([]);
    expect(wrapper.state().favorites).toEqual([]);
  });

  it('Have methods for setting data to state', () => {
    expect(wrapper.state().filmData).toEqual({});
    expect(wrapper.state().peopleData).toEqual([]);
    expect(wrapper.state().planetData).toEqual([]);
    expect(wrapper.state().vehicleData).toEqual([]);

    wrapper.instance().setFilms(cleanFilms);
    wrapper.instance().setPeople(cleanPeople);
    wrapper.instance().setPlanets(cleanPlanets);
    wrapper.instance().setVehicles(cleanVehicles);

    expect(wrapper.state().filmData).toEqual(cleanFilms);
    expect(wrapper.state().peopleData).toEqual(cleanPeople);
    expect(wrapper.state().planetData).toEqual(cleanPlanets);
    expect(wrapper.state().vehicleData).toEqual(cleanVehicles);
  });

  it('Have a click handler which puts clicked objects in state and removes them if they are clicked once more', () => {
    const cardProp = { name: 'Alderaan' };

    expect(wrapper.state().favorites).toEqual([]);

    wrapper.instance().clickHandler(cardProp);

    expect(wrapper.state().favorites).toEqual([cardProp]);

    wrapper.instance().clickHandler(cardProp);

    expect(wrapper.state().favorites).toEqual([]);
  });
});
