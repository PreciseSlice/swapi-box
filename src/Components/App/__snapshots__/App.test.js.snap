// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Exist and matches snapshot 1`] = `
<div
  className="app"
>
  <div
    className="header"
  >
    <h1>
      SWapi-box
    </h1>
  </div>
  <Nav
    favCount={0}
  />
  <div
    className="main"
  >
    <Switch>
      <Route
        exact={true}
        path="/"
        render={[Function]}
      />
      <Route
        path="/people"
        render={[Function]}
      />
      <Route
        path="/planets"
        render={[Function]}
      />
      <Route
        path="/vehicles"
        render={[Function]}
      />
      <Route
        path="/favorites"
        render={[Function]}
      />
    </Switch>
  </div>
</div>
`;
